extensions/amp-playbuzz/0.1/amp-playbuzz.js

```diff
@@ -0,0 +1,290 @@
+    this.displayItemInfo_ = false;
+
+     /** @private {?boolean} */
+    this.displayShareBar_ = false;
+
+     /** @private {?boolean} */
+    this.displayComments_ = false;
+
+     /**  @param {Array.<function>} */
+    this.unlisteners_ = [];
+
+  }
+  /**
+   * @override
+   */
+  preconnectCallback() {
+    this.preconnect.preload(this.item_);
+  }
+
+  /** @override */
+  renderOutsideViewport() {
+    return false;
+  }
+
+  /** @override */
+  buildCallback() {
+    // EXPERIMENT
+    // AMP.toggleExperiment(EXPERIMENT, true); //for dev
+    user().assert(isExperimentOn(this.win, EXPERIMENT),
+      `Enable ${EXPERIMENT} experiment`);
+
+    const e = this.element;
+
+    this.item_ = assertAbsoluteHttpOrHttpsUrl(e.getAttribute('src'));
+    const parsedHeight = parseInt(e.getAttribute('height'), 10);
+
+    this.itemHeight_ = isNaN(parsedHeight) ? this.itemHeight_ : parsedHeight;
+    this.displayItemInfo_ = e.getAttribute('data-item-info') === 'true';
+    this.displayShareBar_ = e.getAttribute('data-share-buttons') === 'true';
+    this.displayComments_ = e.getAttribute('data-comments') === 'true';
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return layout === Layout.RESPONSIVE;
+    // return layout === Layout.CONTAINER;
+    // return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  createPlaceholderCallback() {
+    const placeholder = this.win.document.createElement('div');
+    placeholder.setAttribute('placeholder', '');
+    placeholder.appendChild(this.createPlaybuzzLoader_());
+    return placeholder;
+  }
+
+
+  /**
+   *
+   * Returns the overflow element
+   * @returns {!Element} overflowElement
+   *
+   */
+  getOverflowElement_() {
+    const createElement = utils.getElementCreator(this.element.ownerDocument);
+
+    const overflow = createElement('div', 'pb-overflow');
+    overflow.setAttribute('overflow', '');
+
+    const overflowButton = createElement('button');
+    overflowButton.textContent = 'Show More';
+
+    const arrow = createElement('img', 'pb-arrow-down');
+    arrow.src = showMoreArrow;
+
+    overflowButton.appendChild(arrow);
+    overflow.appendChild(overflowButton);
+
+    return overflow;
+  }
+
+  /** @override */
+  layoutCallback() {
+
+    const iframe = this.element.ownerDocument.createElement('iframe');
+    this.iframe_ = iframe;
+    iframe.setAttribute('frameborder', '0');
+    iframe.setAttribute('allowtransparency', 'true');
+    iframe.setAttribute('allowfullscreen', 'true');
+    iframe.src = this.generateEmbedSourceUrl_();
+
+    this.listenToPlaybuzzItemMessage_('resize_height',
+      utils.debounce(this.itemHeightChanged_.bind(this), 100));
+
+    this.element.appendChild(this.getOverflowElement_());
+
+    this.applyFillContent(iframe);
+    this.element.appendChild(iframe);
+
+    return this.iframePromise_ = this.loadPromise(iframe).then(() => {
+
+      this.attemptChangeHeight(this.itemHeight_).catch(() => {/* die */});
```

Comment:
There's no a lot of reason to call `attemptChangeHeight` here. Normally you only need to update height when your embed sends you a request to do so.

Are you just trying to make sure you have some minimal height here? (300px?) If yes - this code can actually reduce the size of your element in this case.